doc/tutorial/describing-code.rst

```diff
@@ -0,0 +1,226 @@
+Describing code in Sphinx
+=========================
+
+In the :doc:`previous sections of the tutorial </tutorial/index>` you can read
+how to write narrative or prose documentation in Sphinx. In this section you
+will describe code objects instead.
+
+.. _tutorial-describing-objects:
+
+Documenting Python objects
+--------------------------
+
+Sphinx offers several roles and directives to document Python objects,
+all grouped together in the Python
+:doc:`domain </usage/restructuredtext/domains>`. For example, you can
+use the :rst:dir:`py:function` directive to document a Python function,
+as follows:
+
+.. code-block:: rst
+   :caption: docs/source/usage.rst
+
+   Creating recipes
+   ----------------
+
+   To retrieve a list of random ingredients,
+   you can use the ``lumache.get_random_ingredients()`` function:
+
+   .. py:function:: lumache.get_random_ingredients([kind=None])
+
+      Return a list of random ingredients as strings.
+
+      :param kind: Optional "kind" of ingredients.
+      :type kind: list[str] or None
+      :return: The ingredients list.
+      :rtype: list[str]
+
+Which will render like this:
+
+.. figure:: /_static/tutorial/lumache-py-function.png
+   :width: 80%
+   :align: center
+   :alt: HTML result of documenting a Python function in Sphinx
+
+   HTML result of documenting a Python function in Sphinx
```

Comment:
nit: I'd avoid using "HTML result" since this is not the HTML, but the rendered version 😄 

```suggestion
   The rendered result of documenting a Python function in Sphinx
```

Reply:
![treachery](https://user-images.githubusercontent.com/316517/132354487-92db73cf-b790-4e43-86e5-bae2ea3be5c8.jpg)